### DISCUSSION TITLE
Annotated Validator vs Field Validator

### DISCUSSION THREAD

**Discussion by gerhardadler** [pydantic/pydantic#10426] [Question] [answered]
Pydantic has both these solutions for creating validators for specific fields. My understanding is that you can achieve the same behavior using both solutions.

Annotated Validators have the benefit that you can apply the same validator to multiple fields easily, meaning you could have less boiler plate in some situations.

Are there any other reasons to choose one above the other?

**Comment by Viicos** [comment, marked as answer]
As you mentioned, annotated validators can be useful to create reusable fields:

```python
from typing import Annotated

from pydantic import BaseModel, PlainValidator

MyInt = Annotated[int, PlainValidator(...)]

class Model1(BaseModel):
    f1: MyInt

class Model2(BaseModel):
    f2: MyInt  # no code duplication
```

Other than that, some people like having all the information about a field in one place, so that when you look at the type annotation of `f3`:

```python
class Model3(BaseModel):
    f3: Annotated[int, PlainValidator(...)]
    f4: int

    ...

    @field_validator('f4')
    def val(...): ...
``` 

You immediately know what kind of validation is attached to it, contrary to `f4`, where the `field_validator` is defined later in the class body.

**Comment by Jojain** [comment]
I have exactly the same question, does the pydantic team said somewhere what was the prefered way ? 
From code maintainability it seems to me that `Annotated` way is superior. Ideally I'd like to follow common pattern in my code if there is any

**Reply by Viicos** [reply]
I answered [here](https://github.com/pydantic/pydantic/discussions/10426#discussioncomment-10704660).
